src/main/scala/mesosphere/marathon/api/HttpTransferMetricsHandler.scala

```diff
@@ -0,0 +1,130 @@
+package mesosphere.marathon
+package api
+
+import com.typesafe.scalalogging.StrictLogging
+import javax.servlet._
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+import javax.servlet.{AsyncEvent, AsyncListener}
+import javax.ws.rs.core.HttpHeaders
+import mesosphere.marathon.metrics.{Counter, Metrics, ServiceMetric}
+import org.eclipse.jetty.server.Request
+import org.eclipse.jetty.server.handler.AbstractHandler
+import org.eclipse.jetty.server.{HttpChannelState, _}
+
+import HttpTransferMetricsHandler._
+
+/* Container for HTTP Metrics
+ */
+trait HttpTransferMetrics {
+  /**
+    * Metric representing the total HTTP entity gzipped bytes read (does not include HTTP headers)
+    */
+  val gzippedBytesReadMetric: Counter
+
+  /**
+    * Metric representing the total HTTP entity gzipped bytes written (does not include HTTP headers)
+    */
+  val gzippedBytesWrittenMetric: Counter
+
+  /**
+    * Metric representing the total HTTP entity bytes read (does not include HTTP headers)
+    */
+  val bytesReadMetric: Counter
+
+  /**
+    * Metric representing the total HTTP entity bytes written (does not include HTTP headers)
+    */
+  val bytesWrittenMetric: Counter
+}
+
+/**
+  * Implementation class for HttpMetrics
+  *
+  * This class is structured and named this way for backwards compatibility, due to the fact that our metrics are
+  * currently (and lamentably) tied to classes.
+  */
+class HTTPMetricsFilter extends HttpTransferMetrics {
```

Comment:
This is obnoxious. But we've already released the metric with this path? If we decide we can change the metric key then we should fold in to HttpFilter but it seems this would be a breaking change.

Reply:
We released it even in 1.5.x